Console/Command/AbstractStoreCommand.php

```diff
@@ -0,0 +1,128 @@
+<?php
+
+namespace Algolia\AlgoliaSearch\Console\Command;
+
+use Algolia\AlgoliaSearch\Service\StoreNameFetcher;
+use Magento\Framework\App\Area;
+use Magento\Framework\App\State;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Framework\Exception\NoSuchEntityException;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Question\ConfirmationQuestion;
+
+abstract class AbstractStoreCommand extends Command
+{
+    protected const STORE_ARGUMENT = 'store';
+
+    protected ?OutputInterface $output = null;
+    protected ?InputInterface $input = null;
+
+    abstract protected function getCommandName(): string;
+    abstract protected function getCommandDescription(): string;
+    abstract protected function getAdditionalDefinition(): array;
+    abstract protected function getStoreArgumentDescription(): string;
+
+    public function __construct(
+        protected State            $state,
+        protected StoreNameFetcher $storeNameFetcher,
+        ?string                    $name = null
+    )
+    {
+        parent::__construct($name);
+    }
+
+    protected function getCommandPrefix(): string
+    {
+        return 'algolia:';
+    }
+
+    protected function getFullCommandName(): string
+    {
+        return $this->getCommandPrefix() . $this->getCommandName();
+    }
+
+    /**
+     * @inheritDoc
+     */
+    protected function configure(): void
+    {
+        $definition = [$this->getStoreArgumentDefinition()];
+        $definition = array_merge($definition, $this->getAdditionalDefinition());
+
+        $this->setName($this->getFullCommandName())
+            ->setDescription($this->getCommandDescription())
+            ->setDefinition($definition);
+
+        parent::configure();
+    }
+
+    protected function setAreaCode(): void
+    {
+        try {
+            $this->state->setAreaCode(Area::AREA_CRONTAB);
+        } catch (LocalizedException) {
+            // Area code is already set - nothing to do
+        }
+    }
+
+    /**
+     * @param InputInterface $input
+     * @return int[]
+     */
+    protected function getStoreIds(InputInterface $input): array
+    {
+        return (array) $input->getArgument(self::STORE_ARGUMENT);
```

Comment:
Is it possible to check the type of the input from the command ? When I try to run `bin/magento algolia:synonyms:deduplicate foo` , I get an error.

Reply:
Bien sûr ... bonne prise!

I'm falling back on Symfony errors here otherwise the calling class needs to catch and do something with it:
<img width="806" alt="image" src="https://github.com/user-attachments/assets/66ed1d4d-dbf5-4bdc-a2bf-de15ad797652" />